app/scripts/HiGlassComponent.js

```diff
@@ -4053,6 +4054,78 @@ class HiGlassComponent extends React.Component {
     this.setCenters[viewUid](centerX, centerY, k, false, animateTime);
   }
 
+  zoomToGene(viewUid, geneSymbol, animateTime) {
+    if (!(viewUid in this.setCenters)) {
+      throw Error(
+        `Invalid viewUid. Current uuids: ${Object.keys(this.setCenters).join(
+          ',',
+        )}`,
+      );
+    }
+
+    if (
+      !this.state.views[viewUid].genomePositionSearchBox ||
+      !this.state.views[viewUid].genomePositionSearchBox.autocompleteServer ||
+      !this.state.views[viewUid].genomePositionSearchBox.autocompleteId ||
+      !this.state.views[viewUid].chromInfoPath
+    ) {
+      console.warn(
+        'Please set chromInfoPath, autocompleteServer, and autocompleteId to use the zoomToGene API',
+      );
+      return;
+    }
+
+    const autocompleteServer = this.state.views[viewUid].genomePositionSearchBox
+      .autocompleteServer;
+    const autocompleteId = this.state.views[viewUid].genomePositionSearchBox
+      .autocompleteId;
+    const chromInfoPath = this.state.views[viewUid].chromInfoPath;
+
+    const url = `${autocompleteServer}/suggest/?d=${autocompleteId}&ac=${geneSymbol.toLowerCase()}`;
+
+    tileProxy
+      .json(url, toVoid, this.pubSub)
+      .then((positions) => {
```

Comment:
If you wanted to be a little more modern you could use `async / await` but this works perfectly fine too.

Reply:
Thanks for the suggestion! I haven't used `async / await` before, but I will take a look at it!